doc/architecture-decisions/0013-stopped-app-routing.md

```diff
@@ -0,0 +1,101 @@
+# 13. Stopped App Routing
+
+Date: 2020-05-29
+Updated: 2020-06-02
+
+## Status
+
+Proposal
+
+## Context
+
+In CF-for-VMs, when a route is mapped to multiple apps, some started and some
+stopped, gorouter only directs requests to the apps that are started. In the
+case of crashed apps the gorouter also only directs requests to the apps that
+are started and healthy. If **ALL** destinations are stopped or crashed then a
+503 response will occur.
+
+In CF-for-k8s _currently_, we balance traffic equally between the started and
+stopped apps. This results in 503 errors for the user, and is not feature
+parity. [TBD if traffic is equally balanced between crashed app instances and
+healthy app instances]
+
+This is because the components that create the Istio configuration do not
+currently know about the status of apps. They assume all destinations on a Route
+object are up and able to receive traffic. More concretely the Services that are
+created use the app guid and process type as selectors. When an application is
+stopped the StatefulSet and Pods backing that application are deleted. This
+means the Selector can't select any pods and results in a 503.
+
+This ADR is specifically focused on handling routing to stopped applications.
```

Comment:
I'm wondering if it should be the other way around: we first dive into and focus on how to handle the crashed app situation, since that could be viewed as a more extreme version of the stopped app case. If we determine how we plan to avoid sending traffic to crashed app instances, then dealing with the stopped app case is likely pretty simple at that point.